test/e2e/kubectl.go

```diff
@@ -144,13 +145,20 @@ func validateGuestbookApp(c *client.Client, ns string) {
 
 // Returns whether received expected response from guestbook on time.
 func waitForGuestbookResponse(c *client.Client, cmd, arg, expectedResponse string, timeout time.Duration, ns string) bool {
-	for start := time.Now(); time.Since(start) < timeout; time.Sleep(5 * time.Second) {
+	flag := 1
+	expectNoError(wait.Poll(5*time.Second, timeout, func() (bool, error) {
 		res, err := makeRequestToGuestbook(c, cmd, arg, ns)
 		if err == nil && res == expectedResponse {
-			return true
+			flag = 0
+			return true, nil
 		}
+		return false, nil
+	}))
+	if flag == 0 {
```

Comment:
Can you actually get here without flag being zero?  If flag is 1, then you never succeeded, in which case Poll returns an error, expectNoError fails etc.